File: modules/swagger-core/src/test/scala/1_3/converter/CovariantGetterTest.scala
package converter

import com.fasterxml.jackson.module.scala.DefaultScalaModule
import io.swagger.converter.ModelConverters
import io.swagger.util.Json
import matchers.SerializationMatchers._
import models.JCovariantGetter
import org.junit.runner.RunWith
import org.scalatest.{FlatSpec, Matchers}
import org.scalatest.junit.JUnitRunner

@RunWith(classOf[JUnitRunner])
class CovariantGetterTest extends FlatSpec with Matchers {
  val m = Json.mapper()
  m.registerModule(DefaultScalaModule)

  it should "read a getter with covariant return type" in {
    val models = ModelConverters.getInstance().read(classOf[JCovariantGetter.Sub])
    models.size should be(1)
    models should serializeToJson(
      """{
  "Sub" : {
    "type":"object",
    "properties" : {
      "myProperty" : {
        "type" : "integer",
        "format" : "int32",
        "position" : 1
      },
      "myOtherProperty" : {
        "type" : "integer",
        "format" : "int32",
        "position" : 2
      }
    }
  }
}""")
  }
}

File: modules/swagger-core/src/test/java/io/swagger/converting/CovariantGetterTest.java
package io.swagger.converting;

import static org.testng.Assert.assertEquals;

import io.swagger.converter.ModelConverters;
import io.swagger.matchers.SerializationMatchers;

import io.swagger.oas.models.JCovariantGetter;
import io.swagger.v3.oas.models.media.Schema;
import org.testng.annotations.Test;

import java.util.Map;

public class CovariantGetterTest {
    @Test(description = "it should read a getter with covariant return type")
    public void testCovariantGetter() {
        final Map<String, Schema> models = ModelConverters.getInstance().read(JCovariantGetter.Sub.class);
        assertEquals(models.size(), 1);
        final String json = "{" +
                "   \"Sub\":{" +
                "      \"type\":\"object\"," +
                "      \"properties\":{" +
                "         \"myProperty\":{" +
                "            \"type\":\"integer\"," +
                "            \"format\":\"int32\"" +
                "         }," +
                "         \"myOtherProperty\":{" +
                "            \"type\":\"integer\"," +
                "            \"format\":\"int32\"" +
                "         }" +
                "      }" +
                "   }" +
                "}";
        SerializationMatchers.assertEqualsToJson(models, json);
    }
}
